### ISSUE TITLE
out-of-bound-read  In utils/jpgicc/jpgicc.c

### ISSUE THREAD

**Issue by treebacker** [mm2/Little-CMS#255]
In source code `utils/jpgicc/jpgicc.c`, function `ProcessPhotoshopAPP13`.

There is a number maybe negative, which results a `out of bound read`.

As below code shows:

```
 len = ((((GETJOCTET(data[i]<<8) + GETJOCTET(data[i+1]))<<8) +
                         GETJOCTET(data[i+2]))<<8) + GETJOCTET(data[i+3]);

....
 i += len + ((len & 1) ? 1 : 0);   // Alignment
```

`len` may be a negative number with a crafted file, then `i` may be out of bound.

command line: ./utils/jpgicc/.libs/jpgicc bug1 /dev/null

in AddressSanitizer:
![1](https://user-images.githubusercontent.com/44844446/115103417-45c57500-9f84-11eb-8b2c-e6ddad8cc308.png)
In gdb:
![bug1_1](https://user-images.githubusercontent.com/44844446/115103424-55dd5480-9f84-11eb-8cea-eaaa917808ad.png)
the reason for `i may be out of bound`:
![bug1_2](https://user-images.githubusercontent.com/44844446/115103425-5d9cf900-9f84-11eb-9ba6-49fe923e0ff4.png)

The crafted file is in the zip:
[bug1.zip](https://github.com/mm2/Little-CMS/files/6328685/bug1.zip)

**Comment by mm2** [comment]
Thanks for reporting.
However, what you are reporting may be a vulnerability of jpegicc, which is a **sample** of how to use the little CMS library and **it is not required or used** in client applications using the library. So, I will try to add some checks as long as that would not affect code readibility, which is the primary goal of this sample.

 **Please note any CVE regarding this kind of vulnerabilities would be promply disputed, as are related with samples and not with the little cms component, which is unaffected**.

**Comment by mm2** [comment]
Fixed per 8564c9b4fbf07d23062bbbbd87d5eb10dd92073f Thanks for reporting
